File: Array/print_duplicate.rb
def print_deplicates(arr)
  max = arr.length - 1
  0.upto(max) do |i|
    if arr[arr[i].abs] > 0
      arr[arr[i].abs] = -arr[arr[i].abs]
    # else
      # puts arr.to_s
    end
  end
  arr.each_with_index do |element, indx|
    print "#{indx} " if element < 0
  end
end

arr = [1, 3, 3, 1, 3, 6, 6]
print_deplicates(arr)
puts

File: Array/print_duplicate.py
def print_deplicates(arr):
  high = len(arr)
  for i in range(high):
    ele = arr[i] % high
    arr[ele] += high
  print("Arr: {}".format(arr))

  for i in range(high):
    ele = arr[i] % high
    if arr[ele] >= high * 2 :
      print(ele, end = ' ')
    arr[ele] = arr[ele] % high
  print()

arr = [1, 6, 3, 1, 2, 3, 6, 3]
arr = [1, 3, 3, 1, 1, 1, 3, 6, 6, 0,2]

print("Arr: {}".format(arr))
print_deplicates(arr)
